app/models/manageiq/providers/container_manager/container_template.rb

```diff
@@ -0,0 +1,2 @@
+class ManageIQ::Providers::ContainerManager::ContainerTemplate < ContainerTemplate
```

Comment:
Adding this (now empty) subclass had an interesting consequence (that's breaking openshift refresh).
In [manageiq/providers/container_manager.rb](https://github.com/ManageIQ/manageiq/blob/c70bc536d871a68bcfdf50c9a033ded8bc1330c5/app/models/manageiq/providers/container_manager.rb#L25),
`ContainerTemplate` resolves to this nested class, not the base `::ContainerTemplate`.
And when we do
```
    has_many :container_templates, :foreign_key => :ems_id, :dependent => :destroy
```
it apparently does such a const lookup (?), and the query now constrains `type`, never matching any records:
```
(byebug) ems.container_templates.to_sql
"SELECT \"container_templates\".* FROM \"container_templates\" WHERE \"container_templates\".\"type\" IN ('ManageIQ::Providers::ContainerManager::ContainerTemplate') AND \"container_templates\".\"ems_id\" = 24000000000002"
```

I can do
```
    has_many :container_templates, :foreign_key => :ems_id, :dependent => :destroy, :class_name => "::ContainerTemplate"
```
to get back the query
```
> ems.container_templates.to_sql
=> "SELECT \"container_templates\".* FROM \"container_templates\" WHERE \"container_templates\".\"ems_id\" = 3"
```
but I wonder what's the point of having both `::ContainerTemplate` and `ManageIQ::Providers::ContainerManager::ContainerTemplate`?
It's not like any non-container providers will have ContainerTemplate, so why not axe this subclass and have kubernetes/openshift inherit from `::ContainerTemplate`?

@lfu @Fryguy @kbrock @moolitayer please advice.